### PR TITLE
Define public API for casting

### PR DESCRIPTION
**Pull request by rolandkakonyi**
## Description

This PR defines the public API regarding the player and events, related for the Casting Feature for Android and iOS.

## Changes

- Added the player APIs needed for the casting feature.
- Added casting related events.

## Checklist
- [x] 🗒 `CHANGELOG` entry

### CODE REVIEW

File: src/events.ts [hunk 1]
```diff
@@ -554,3 +554,83 @@ export interface VideoPlaybackQualityChangedEvent extends Event {
...
+ * Emitted when casting to a cast-compatible device is available.
+ */
+export interface CastAvailableEvent extends Event {}
+
+/**
+ * Emitted when the playback on a cast-compatible device was paused.
+ *
+ * On Android `PausedEvent` is also emitted while casting.
+ */
+export interface CastPaused extends Event {}
```
Comment (strangesource) [inline src/events.ts:568]:
```suggestion
export interface CastPausedEvent extends Event {}
```
For consistency with existing event naming. Holds for all events below.

File: src/player.ts [hunk 2]
```diff
@@ -491,4 +496,42 @@ export class Player extends NativeInstance<PlayerConfig> {
...
+   * should be sent.
+   *
+   * @platform iOS, Android
+   */
+  castVideo = () => {
+    PlayerModule.castVideo(this.nativeId);
+  };
+
+  /**
+   * Stops casting the current video. Has no effect if [isCasting] is false.
```
Comment (strangesource) [inline src/player.ts:530]:
```suggestion
   * Stops casting the current video. Has no effect if `isCasting` is false.
```
I don't think `[]` linking is supported in typescript

File: src/remoteControlConfig.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  /**
+   * A Map containing custom configuration values that are sent to the remote control receiver.
+   * Default value is an empty map.
+   */
+  customReceiverConfig?: Record<string, string | null>;
+  /**
+   * Whether casting is enabled.
+   * Default value is `true`.
+   *
+   * Has no effect if the BitmovinCastManager is not initialized in the app's context.
```
Comment (strangesource) [inline src/remoteControlConfig.ts:19]:
```suggestion
   * Has no effect if the `BitmovinCastManager` is not initialized in the app's context.
```
Does that make sense for iOS as well?

**Review by strangesource — COMMENTED** [review]
Looking good in general, just some minor things/questions.

File: src/events.ts (hunk 1 above)
Comment (rolandkakonyi) [reply src/events.ts]: Oops, I slipped on the multi-cursor edit.
Fixed in 56ee8d3

File: src/player.ts (hunk 2 above)
Comment (rolandkakonyi) [reply src/player.ts]: Fixed in 47a8071

File: src/remoteControlConfig.ts (hunk 3 above)
Comment (rolandkakonyi) [reply src/remoteControlConfig.ts]: Improved in 1d77a80

**Review by strangesource — APPROVED** [review]
Nice work!

**Pull request merged**